diff -urN tftp-hpa-0.48/CHANGES tftp-hpa-0.48-inq/CHANGES
--- tftp-hpa-0.48/CHANGES	Wed Jan 31 02:51:05 2007
+++ tftp-hpa-0.48-inq/CHANGES	Thu Mar 13 20:26:29 2008
@@ -1,3 +1,7 @@
+Changes in 0.48-inquisitor:
+	Addition log notice "SENT" that tells us about file's
+	transmission finishing.
+
 Changes in 0.48:
 	Unbreak -l -s in the server, which was broken in 0.47.
 
diff -urN tftp-hpa-0.48/tftpd/tftpd.c tftp-hpa-0.48-inq/tftpd/tftpd.c
--- tftp-hpa-0.48/tftpd/tftpd.c	Wed Jan 31 02:51:05 2007
+++ tftp-hpa-0.48-inq/tftpd/tftpd.c	Thu Mar 13 20:27:11 2008
@@ -820,15 +820,21 @@
   }
   
   if ( ap != (ackbuf+2) ) {
-    if ( tp_opcode == WRQ )
+    if ( tp_opcode == WRQ ) {
       (*pf->f_recv)(pf, (struct tftphdr *)ackbuf, ap-ackbuf);
-    else
+    } else {
       (*pf->f_send)(pf, (struct tftphdr *)ackbuf, ap-ackbuf);
+      syslog(LOG_NOTICE, "SENT %s to %s\n",
+        filename, inet_ntoa(from.sin_addr));
+    }
   } else {
-    if (tp_opcode == WRQ)
+    if (tp_opcode == WRQ) {
       (*pf->f_recv)(pf, NULL, 0);
-    else
+    } else {
       (*pf->f_send)(pf, NULL, 0);
+      syslog(LOG_NOTICE, "SENT %s to %s\n",
+        filename, inet_ntoa(from.sin_addr));
+    }
   }
   exit(0);			/* Request completed */
 }
